api/v1alpha1/snapshot_types.go

```diff
@@ -6,30 +6,40 @@
 package v1alpha1
 
 import (
-	"strings"
+	"fmt"
 
+	"github.com/mitchellh/hashstructure/v2"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// Identity defines a cache entry. It is used to generate a hash that is then used by the
+// caching layer to identify an entry.
+type Identity map[string]string
+
+func (i *Identity) Hash() (string, error) {
+	hash, err := hashstructure.Hash(i, hashstructure.FormatV2, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to hash identity: %w", err)
+	}
+
+	return fmt.Sprintf("sha-%d", hash), nil
+}
+
 // SnapshotSpec defines the desired state of Snapshot
 type SnapshotSpec struct {
-	Ref string `json:"ref"`
+	Identity Identity `json:"identity"`
 }
 
 // SnapshotStatus defines the observed state of Snapshot
 type SnapshotStatus struct {
 	// +optional
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 
-	// +optional
-	RepositoryURL string `json:"repositoryURL,omitempty"`
```

Comment:
It still would be nice to keep this, so consumers don't have to know how to compute it. Let's say I want to manually verify that the image exist.

Reply:
👍